### ISSUE TITLE
Error: ambiguous config file format, config inside hidden folder

### ISSUE THREAD

**Issue by victordidenko** [caddyserver/caddy#6373]
Similar to https://github.com/caddyserver/caddy/issues/6363 after update caddy doesn't start with error

```
2024/06/04 10:49:48.000	INFO	using config from file	{"file": ".serve/Caddyfile.preview"}
Error: ambiguous config file format; please specify adapter (use --adapter)
```

```sh
$ caddy version
v2.8.4 h1:q3pe0wpBj1OcHFZ3n/1nl4V4bxBrYoSoab7rL9BMYNk=
```

Note that I use config from subfolder `.serve`

Is it possible that `filepath.Base` function does not handle that case correctly?
https://github.com/omalk98/caddy/blob/3f55efcfde09bc3d0a56597b1280dd2289805f69/cmd/main.go#L150

**Comment by francislavoie** [comment]
Being fixed here https://github.com/caddyserver/caddy/pull/6370

For now add `--adapter caddyfile`

**Comment by victordidenko** [comment]
Yes, with `--adapter caddyfile` it is working, sorry didn't mention this in the first message

**Comment by mholt** [comment]
Sorry for the regression here. Will be fixed momentarily with #6370
